internal/helm-storage-backend/template.go

```diff
@@ -2,32 +2,214 @@ package helmstoragebackend
+	_, err := h.GetValue(ctx, &pb.GetValueRequest{
+		TypeInstanceId: req.TypeInstanceId,
+		Context:        req.Context,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	return &pb.OnCreateResponse{}, nil
+}
+
+// OnUpdate only validates if provided goTemplate can be later rendered on GetValue request.
+func (h *TemplateHandler) OnUpdate(ctx context.Context, req *pb.OnUpdateRequest) (*pb.OnUpdateResponse, error) {
+	h.log.Info("updating entry", zap.String("id", req.TypeInstanceId))
+	// dry-run that get will work
+	_, err := h.GetValue(ctx, &pb.GetValueRequest{
+		TypeInstanceId: req.TypeInstanceId,
+		Context:        req.Context,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	h.log.Info("return entry up")
+	return &pb.OnUpdateResponse{}, nil
+}
+
+// OnDelete does nothing.
+func (*TemplateHandler) OnDelete(context.Context, *pb.OnDeleteRequest) (*pb.OnDeleteResponse, error) {
+	return &pb.OnDeleteResponse{}, nil
+}
+
+// GetLockedBy does nothing.
+func (*TemplateHandler) GetLockedBy(context.Context, *pb.GetLockedByRequest) (*pb.GetLockedByResponse, error) {
+	return &pb.GetLockedByResponse{}, nil
+}
+
+// OnLock does nothing.
+func (*TemplateHandler) OnLock(context.Context, *pb.OnLockRequest) (*pb.OnLockResponse, error) {
+	return &pb.OnLockResponse{}, nil
+}
+
+// OnUnlock does nothing.
+func (*TemplateHandler) OnUnlock(context.Context, *pb.OnUnlockRequest) (*pb.OnUnlockResponse, error) {
+	return &pb.OnUnlockResponse{}, nil
+}
+
+func (h *TemplateHandler) getReleaseContext(contextBytes []byte) (*TemplateContext, error) {
+	var ctx TemplateContext
+	err := json.Unmarshal(contextBytes, &ctx)
+	if err != nil {
+		return nil, gRPCInternalError(errors.Wrap(err, "while unmarshaling context"))
+	}
+
+	if ctx.HelmRelease.Driver == nil {
+		ctx.HelmRelease.Driver = ptr.String(defaultHelmDriver)
+	}
+
+	return &ctx, nil
+}
+
+func (h *TemplateHandler) fetchHelmRelease(ti string, ctx []byte) (*release.Release, *TemplateContext, error) {
+	relCtx, err := h.getReleaseContext(ctx)
+	if err != nil {
+		return nil, nil, gRPCInternalError(err)
+	}
+
+	rel, err := h.fetcher.FetchHelmRelease(ti, relCtx.HelmRelease)
+	if err != nil {
+		return nil, nil, err // it already handles grpc errors properly
+	}
+
+	return rel, relCtx, nil
+}
+
+func (h *TemplateHandler) renderOutputValue(template string, helmRelease *release.Release) ([]byte, error) {
+	if helmRelease.Chart == nil {
+		return nil, gRPCInternalError(errors.New("Helm release doesn't have associated Helm chart"))
+	}
+	args := helm.OutputArgs{
+		GoTemplate: template,
+	}
+
+	// It is important to load the chart dependencies as by default they are not
+	// available and if chart was using that we may miss some value and template funcs.
+	chartDeps, err := h.loadHelmChartDependencies(helmRelease.Chart)
+	if err != nil {
+		return nil, err
+	}
+
+	helmRelease.Chart.SetDependencies(chartDeps...)
+	if err := chartutil.ProcessDependencies(helmRelease.Chart, helmRelease.Config); err != nil {
+		return nil, err
```

Comment:
I think we should wrap the error with `gRPCInternalError` in these places as well